platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py

```diff
@@ -2346,3 +2369,223 @@ def get_error_description(self):
         else:
             error_description = "Unknow SFP module status ({})".format(oper_status)
         return error_description
+
+
+class RJ45Port(SFP):
+    """class derived from SFP, representing RJ45 ports"""
+
+    def __init__(self, sfp_index):
+        super(RJ45Port, self).__init__(sfp_index, RJ45_TYPE)
+        self._sfp_type = RJ45_TYPE
+
+    # Reuse sdk_handle
+
+    @property
+    def sfp_type(self):
+        return self._sfp_type
+
+    def get_presence(self):
+        """
+        Retrieves the presence of the device
+
+        Returns:
+            bool: True if device is present, False if not
+        """
+        # Reimplement it via using SDK API
+        return True
```

Comment:
how can return true is taking from the sdk api if it is not being called? i believe the comment should be that it is assumed that it is always present but then i dont understand. why it is assumed to be always present? is this what we eventually agreed with Ortal? arent you suppose to read it somehow?

Reply:
yes, for RJ45 `get_presence` always returns True. `TRANSCEIVER_STATUS|error_status` will be leveraged for representing `not present` and `unknown` states.
This is because:
1. According to arch spec, there are 3 states for RJ45: `present`, `not present`, `unknown`.
   However, get_presence is boolean and can represent only 2 states. Changing it requires huge effort
2. This is to minimize the code change for RJ45. By doing so xcvrd doesn't need to be changed and utilities will be simple to change.